kotest-assertions/src/commonMain/kotlin/io/kotest/matchers/regex/RegexMatchers.kt

```diff
@@ -0,0 +1,140 @@
+
+import io.kotest.Matcher
+import io.kotest.MatcherResult
+import io.kotest.should
+import io.kotest.shouldNot
+
+fun haveSamePatternMatcher(pattern: String) = object : Matcher<Regex> {
+   override fun test(value: Regex): MatcherResult {
+      return MatcherResult(
+         value.pattern == pattern,
+         { "Regex should have pattern ${pattern} but has pattern ${value.pattern}" },
+         { "Regex should not have pattern ${value.pattern}" }
+      )
+   }
+}
+
+fun haveSameRegexOptionsMatcher(options: Set<RegexOption>) = object : Matcher<Regex> {
+   override fun test(value: Regex): MatcherResult {
+      return MatcherResult(
+         value.options == options,
+         { "Regex should have options ${options} but has options ${value.options}" },
+         { "Regex should not have pattern ${value.options}" }
+      )
+   }
+}
+
+fun haveExactRegexOptionMatcher(option: RegexOption) = object : Matcher<Regex> {
+   override fun test(value: Regex): MatcherResult {
+      return MatcherResult(
+         value.options.contains(option),
+         { "Regex options should contains $option" },
+         { "Regex options should not contains $option" }
+      )
+   }
+}
+
+fun haveRegexOptionMatcher(options: Set<RegexOption>) = object : Matcher<Regex> {
+   override fun test(value: Regex): MatcherResult {
+      return MatcherResult(
+         value.options.containsAll(options),
+         { "Regex options should contains $options, but missing ${options.filterNot { value.options.contains(it) }}." },
+         { "Regex options should not contains $options, but containing ${options.filter { value.options.contains(it) }}." }
+      )
+   }
+}
+
+fun areEqualRegexMatcher(regex: Regex) = object : Matcher<Regex> {
+   override fun test(value: Regex): MatcherResult {
+      val patternMatchingResult = haveSamePatternMatcher(regex.pattern).test(value)
+      val optionMatchingResult = haveSameRegexOptionsMatcher(regex.options).test(value)
+      return MatcherResult(
+         patternMatchingResult.passed() && optionMatchingResult.passed(),
+         { "Regex should have pattern ${regex.pattern} and regex options ${regex.options}, but has pattern ${value.pattern} and regex options ${value.options}." },
+         { "Regex should not have pattern ${value.pattern} and regex options ${value.options}." }
+      )
+   }
+}
+
+fun beRegex(regex: Regex) = areEqualRegexMatcher(regex)
+
+fun havePatter(pattern: String) = haveSamePatternMatcher(pattern)
+
+fun haveExactOptions(options: Set<RegexOption>) = haveSameRegexOptionsMatcher(options)
+
+fun haveOption(option: RegexOption) = haveExactRegexOptionMatcher(option)
+
+fun haveOptions(options: Set<RegexOption>) = haveRegexOptionMatcher(options)
+
+/**
+ * Assert that [Regex] is equal to [anotherRegex] by comparing their pattern and options([RegexOption]).
+ * @see [shouldNotBeRegex]
+ * @see [beRegex]
+ * */
+infix fun Regex.shouldBeRegex(anotherRegex: Regex) = this should beRegex(anotherRegex)
```

Comment:
Can we place the infix versions one line above the non infix so it's easy to see which have infix and which dont ?